client/gutenberg/extensions/simple-payments/edit.js

```diff
@@ -163,164 +137,35 @@ class SimplePaymentsEdit extends Component {
-			} );
-			return false;
-		}
-
-		if ( parseFloat( price ) < 0 ) {
-			this.setState( {
-				fieldPriceError: __(
-					'Your price is negative — enter a positive number so people can pay the right amount.'
-				),
-			} );
-			return false;
-		}
-
-		if ( this.decimalPlaces( price ) > precision ) {
-			if ( precision === 0 ) {
-				this.setState( {
-					fieldPriceError: __(
-						'We know every penny counts, but prices can’t contain decimal values.'
-					),
-				} );
-				return false;
-			}
-
-			this.setState( {
-				fieldPriceError: sprintf(
-					_n(
-						'The price cannot have more than %d decimal place.',
-						'The price cannot have more than %d decimal places.',
-						precision
-					),
-					precision
-				),
-			} );
-			return false;
-		}
-
-		if ( this.state.fieldPriceError ) {
-			this.setState( { fieldPriceError: null } );
-		}
-
-		return true;
-	};
-
-	/**
-	 * Validate email
-	 *
-	 * Stores error message in state.fieldEmailError
-	 *
-	 * @returns {Boolean} True when valid, false when invalid
-	 */
-	validateEmail = () => {
-		const { email } = this.props.attributes;
-		if ( ! email ) {
-			this.setState( {
-				fieldEmailError: __(
-					'We want to make sure payments reach you, so please add an email address.'
-				),
-			} );
-			return false;
-		}
-
-		if ( ! emailValidator.validate( email ) ) {
-			this.setState( {
-				fieldEmailError: sprintf( __( '%s is not a valid email address.' ), email ),
-			} );
-			return false;
-		}
-
-		if ( this.state.fieldEmailError ) {
-			this.setState( { fieldEmailError: null } );
-		}
-
-		return true;
-	};
-
-	/**
-	 * Validate title
-	 *
-	 * Stores error message in state.fieldTitleError
-	 *
-	 * @returns {Boolean} True when valid, false when invalid
-	 */
-	validateTitle = () => {
-		const { title } = this.props.attributes;
-		if ( ! title ) {
-			this.setState( {
-				fieldTitleError: __(
-					'Please add a brief title so that people know what they’re paying for.'
-				),
-			} );
-			return false;
-		}
-
-		if ( this.state.fieldTitleError ) {
-			this.setState( { fieldTitleError: null } );
-		}
-
-		return true;
-	};
-
 	handleEmailChange = email => {
+		this.props.setFieldValue( 'email', email );
 		this.props.setAttributes( { email } );
-		this.setState( { fieldEmailError: null } );
 	};
 
 	handleContentChange = content => {
+		this.props.setFieldValue( 'content', content );
 		this.props.setAttributes( { content } );
 	};
 
 	handlePriceChange = price => {
-		price = parseFloat( price );
-		if ( ! isNaN( price ) ) {
-			this.props.setAttributes( { price } );
-		} else {
-			this.props.setAttributes( { price: undefined } );
-		}
-		this.setState( { fieldPriceError: null } );
+		this.props.setFieldValue( 'price', price );
+		this.props.setAttributes( { price: parseFloat( price ) } );
 	};
 
 	handleCurrencyChange = currency => {
+		this.props.setFieldValue( 'currency', currency );
 		this.props.setAttributes( { currency } );
+		this.props.setFieldTouched( 'price' );
```

Comment:
This is so that when the user has a price that contains decimals, we show the error "no decimals allowed" when changing currency to HUF or JPY.